flow/Promise.js

```diff
@@ -33,6 +33,10 @@ declare class Promise<+R> {
   static race<T>(promises: Array<Promise<T>>): Promise<T>;
 
   // Non-standard APIs
+  finally<U>(
+    onSettled?: ?(value: any) => Promise<U> | U
```

Comment:
Name this onFinally, like the spec. Also remove the parameter and set the return type to void.